building-and-packaging/material/packaging_python_slides.md

```diff
@@ -0,0 +1,294 @@
+- Do not use `sudo pip install`! Various security issues with doing so. Go for `pip install --user <package-name>`
+
+---
+
+## What is PyPI?
+
+- [PyPI](https://pypi.org/) = **The Python Package Index** is a repository of software developed in the Python community
+- PyPI itself is developed on GitHub through another software called [Warehouse](https://github.com/pypa/warehouse)
+- PyPI has a informative [public dashboard](https://p.datadoghq.com/sb/7dc8b3250-85dcf667bd) to show its activity
+- A major advantage is the active maintenance of PyPI and the packages indexed in it
+- Not to be confused with **PyPA** which is Python Packaging Authority, a working group which maintains projects in Python packaging
+
+---
+
+## Example of PyPI package: fenicsprecice
+
+- Having a look at [fenicsprecice](https://pypi.org/project/fenicsprecice/)
+
+---
+
+## Python Enhancement Proposals (PEPs)
+
+- PEP is an evolving design document which provides information regarding new features of Python, new processes and new environments
+- PEPs typically involve concise technical information, which also acts as standardizations
+- Packaging workflows are also standardized through PEPs. Example:
+    - [PEP 427](https://www.python.org/dev/peps/pep-0427/) introduces the built-package format "wheel"
+- It is always a good idea to look for **PEP** conventions before trying out processes
+
+---
+
+## File structure for code packaging 1/2
+
+- Most essential component of packaging is file structure. Basic file structure is as follows:
+
+```bash
+generic_folder_name/
+└── src/
+    └── package_name/
+        ├── __init__.py
+        └── solver.py
+```
+
+- The file `__init__.py` is required to import the `package_name/` as a package. This file is mostly empty
+- `solver.py` will contain the code developed by the user
+
+---
+
+## File structure for code packaging 2/2
+
+- Once the file structuring is complete, the repository will look like:
+
+```bash
+generic_folder_name/
+├── LICENSE
+├── setup.py
+├── README.md
+├── src/
+│   └── package_name/
+│       ├── __init__.py
+│       └── solver.py
+└── tests/
+```
+
+<span>
+Lets have a detailed look at individual files in this folder structure
+<!-- .element: class="fragment" data-fragment-index="1" --></span>
+
+---
+
+## Additional options in setup.py 1/2
+
+- [Classifiers](https://pypi.org/classifiers/): additional metadata for the version of the package
+- Defined as part of [PEP 303](https://www.python.org/dev/peps/pep-0301/#distutils-trove-classification)
+- Example:
+
+```bash
+from setuptools import setup
+
+setup(
+    ...
+    classifiers=[
+        "Programming Language :: Python :: <Python-version>",
+        "License :: OSI Approved :: <Name of License>",
+        "Operating System :: <OS's on which this package runs>",
+    ],
+    ...
+)
+```
+
+---
+
+## Additional options in setup.py 2/2
+
+- The file `README.md` can be passed as a long description in the following way:
```

Comment:
I would (maybe) move this part to the README slide or to the beginning when talking about the `setup.py`? 

At the moment with this slide and the README slide being so close to each other it feels a lot like repetition. The README slide also mentions that one includes the READMA as longer description.